Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -149,6 +191,16 @@ void Serializer::EncodeFieldValue(const FieldValue& field_value,
       EncodeInteger(&stream, field_value.integer_value());
       break;
 
+    case FieldValue::Type::String:
+      status = pb_encode_tag(&stream, PB_WT_STRING,
+                             google_firestore_v1beta1_Value_string_value_tag);
+      if (!status) {
+        // TODO(rsgowman): figure out error handling
```

Comment:
Rather than do all that refactoring, I'd probably be better off to just implement error handling; it really isn't going to be all that complicated. I'll issue a followup PR.